pkg/api/cluster_client.go

```diff
@@ -63,12 +64,12 @@ func (c ClusterClient) Read(ctx context.Context, id string) (*models.Cluster, er
 	return &response.Data, err
 }
 
-func (c ClusterClient) ReadByName(ctx context.Context, name string) (*models.Cluster, error) {
+func (c ClusterClient) ReadByName(ctx context.Context, projectId, name string) (*models.Cluster, error) {
 	clusters := struct {
 		Data []models.Cluster `json:"data"`
 	}{}
 
-	url := fmt.Sprintf("clusters?name=%s", name)
+	url := fmt.Sprintf("projects/%s/clusters?name=%s", projectId, name)
```

Comment:
is it really this easy?

Reply:
Yeah. Actually, the API v3 and API v2 outputs are the same, just the endpoints are changing. 
I didn't find anything else 🤷 . Anyhow, we need to test this branch thoroughly.